syntax/syntax.go

```diff
@@ -9,6 +9,33 @@ package syntax
 type Node interface {
 	// Span returns the start and end position of the expression.
 	Span() (start, end Position)
+
+	Comment() *Comments
+}
+
+// A Comment represents a single # comment.
+type Comment struct {
+	Start  Position
+	Token  string // without trailing newline
+	Suffix bool   // an end of line (not whole line) comment
+}
+
+// Comments collects the comments associated with an expression.
+type Comments struct {
+	Before []Comment // whole-line comments before this expression
+	Suffix []Comment // end-of-line comments after this expression
```

Comment:
Can there be more than one of these? I assume not.

Add "(up to 1)" to the doc comment.

Reply:
Not yet done.